docs/using.rst

```diff
@@ -655,8 +612,106 @@ The following files are available:
+refer to config-file_.
+
+If a hook exits with a non-zero exit code, the error will be printed
+to ``stderr`` but renewal will be attempted anyway. A failing hook
+doesn't directly cause Certbot to exit with a non-zero exit code, but
+since Certbot exits with a non-zero exit code when renewals fail, a
+failed hook causing renewal failures will indirectly result in a
+non-zero exit code. Hooks will only be run if a certificate is due for
+renewal, so you can run the above command frequently via ``certbot renew`` 
+without unnecessarily stopping your webserver.
+
+Unique ``--pre-hook`` and ``--post-hook`` hooks run before and after every renewal
+attempt. If two certificates have identical pre and post hooks, they will be run
+only once. Meaning renewal for the two certificates would follow this form:
+pre-hook is run, renewal for cert 1 is run, renewal for cert 2 is run, post-hook
+is run.
+
+Note that certificates outside of the renewal window (30 days) will not have
+renewal attempted, and will therefore not cause pre and post hooks to be run.
+If you want your hook to run only after a successful renewal, use
+``--deploy-hook`` in a command like this.
+
+``certbot renew --deploy-hook /path/to/deploy-hook-script``
+
+For example, if you have a daemon that does not read its certificates as the
+root user, a deploy hook like this can copy them to the correct location and
+apply appropriate file permissions.
+
+/path/to/deploy-hook-script
+
+.. code-block:: none
+
+   #!/bin/sh
+
+   set -e
+
+   for domain in $RENEWED_DOMAINS; do
+           case $domain in
+           example.com)
+                   daemon_cert_root=/etc/some-daemon/certs
+
+                   # Make sure the certificate and private key files are
+                   # never world readable, even just for an instant while
+                   # we're copying them into daemon_cert_root.
+                   umask 077
+
+                   cp "$RENEWED_LINEAGE/fullchain.pem" "$daemon_cert_root/$domain.cert"
+                   cp "$RENEWED_LINEAGE/privkey.pem" "$daemon_cert_root/$domain.key"
+
+                   # Apply the proper file ownership and permissions for
+                   # the daemon to read its certificate and key.
+                   chown some-daemon "$daemon_cert_root/$domain.cert" \
+                           "$daemon_cert_root/$domain.key"
+                   chmod 400 "$daemon_cert_root/$domain.cert" \
+                           "$daemon_cert_root/$domain.key"
+
+                   service some-daemon restart >/dev/null
+                   ;;
+           esac
+   done
+
+If you would like to change what hooks are being run during the renewal of a
+certificate you can either request that certificate again and specify all of
+the arguments you want to be associated with it (hooks and otherwise), or
+manually edit its renewal configuration file. For information on the renewal
+configuration file please refer to config-file_.
```

Comment:
s/file/files

Also, this links to the section on INI configuration files which presumably isn't what you wanted. For your convenience, there's documentation about how to do this [here](http://www.sphinx-doc.org/en/stable/markup/inline.html#ref-role).